test/hooks/common.js

```diff
@@ -45,32 +37,66 @@ var SERVER_RES = '1729';
 var SERVER_KEY = fs.readFileSync(path.join(__dirname, 'fixtures', 'key.pem'));
 var SERVER_CERT = fs.readFileSync(path.join(__dirname, 'fixtures', 'cert.pem'));
 
+function init(agent) {
+  agent._shouldTraceArgs = [];
+  var shouldTrace = agent.shouldTrace;
+  agent.shouldTrace = function() {
+    agent._shouldTraceArgs.push([].slice.call(arguments, 0));
+    return shouldTrace.apply(this, arguments);
+  };
+}
+
 /**
  * Cleans the tracer state between test runs.
  */
-function cleanTraces() {
+function cleanTraces(agent) {
+  if (arguments.length !== 1) {
```

Comment:
Will these `argument` length checks only be around until you're sure all `common` functions have been substituted correctly?

Also, checking the shape of `arguments[0]` may be more effective here...